components/ChallengeMaterial.tsx

```diff
@@ -139,35 +147,137 @@ export const ChallengeTitleCard: React.FC<ChallengeTitleCardProps> = ({
   )
 }
 
-type ChallengeQuestionCardProps = {
+const ChallengeQuestionCardDisplay: React.FC<{
   currentChallenge: ChallengeSubmissionData
-}
-export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
-  currentChallenge
-}) => {
-  const diff = _.get(currentChallenge, 'submission.diff', '')
-  const comment = _.get(currentChallenge, 'submission.comment', '')
-  const updatedAt = _.get(currentChallenge, 'submission.updatedAt', Date.now())
+}> = ({ currentChallenge }) => {
+  const comment = currentChallenge.submission?.comment
+  const updatedAt = currentChallenge.submission?.updatedAt || ''
   const context = useContext(GlobalContext)
   const name = context.session?.user?.name
   const username = context.session?.user?.username
-  const reviewerUserName = _.get(
-    currentChallenge,
-    'submission.reviewer.username',
-    null
-  )
+  const userId = context.session?.user?.id
+
+  const [index, setIndex] = useState(-1)
+
+  const [submission, setSubmission] = useState(currentChallenge.submission)
+  const reviewerUserName = submission?.reviewer?.username || null
   const comments =
-    currentChallenge.submission &&
-    currentChallenge.submission.comments?.filter(comment => !comment.line)
+    submission && submission.comments?.filter(comment => !comment.line)
 
   const [commentValue, setCommentValue] = React.useState('')
   const [addComment] = useAddCommentMutation()
-  const update = updateCache(
-    currentChallenge.submission?.id!,
-    commentValue,
-    name!,
-    username!
-  )
+
+  const { data, loading, error } = useGetPreviousSubmissionsQuery({
+    variables: {
+      challengeId: currentChallenge.submission?.challengeId!,
+      userId: userId!
+    },
+    skip: userId === undefined || currentChallenge.submission === undefined
+  })
+  useEffect(() => {
+    if (data?.getPreviousSubmissions) {
+      if (data.getPreviousSubmissions[index])
+        setSubmission(data.getPreviousSubmissions[index] as Submission)
+      else
+        setSubmission(
+          data.getPreviousSubmissions[
+            data.getPreviousSubmissions.length - 1
+          ] as Submission
```

Comment:
could use lodash here and just do `_.last(data.getPreviousSubmissions) as Submission`

Reply:
Done. I wish javascript supported negative indexes. `array[ -1 ]` would be so much better her.